include/cuco/detail/hash_functions/identityhash.cuh

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuco/detail/hash_functions/utils.cuh>
+#include <cuco/extent.cuh>
+
+#include <cstddef>
+#include <cstdint>
+
+namespace cuco::detail {
+
```

Comment:
I feel the whole `IdentityHash_*` logic can be just 
```cpp
template<typename Key>
using identity_hash = thrust::identity<Key>;
```
Make sense to you? Probably no need to have `_32`/`_64` variants since cuco internals will convert them to `size_type` anyway.

Reply:
Broken out into [PR 514]( https://github.com/NVIDIA/cuCollections/pull/514)

Reply:
If this is in another PR then it is, for this PR, resolved, yes?